kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/data/IcebergWriterFactory.java

```diff
@@ -112,4 +118,18 @@ Table autoCreateTable(String tableName, SinkRecord sample) {
             });
     return result.get();
   }
+
+  @VisibleForTesting
+  static void createNamespaceIfNotExist(Catalog catalog, Namespace identifierNamespace) {
+    String[] levels = identifierNamespace.levels();
+    for (int index = 0; index < levels.length; index++) {
+      Namespace namespace = Namespace.of(Arrays.copyOfRange(levels, 0, index + 1));
+      try {
+        ((SupportsNamespaces) catalog).createNamespace(namespace);
```

Comment:
This can cause a ClassCastException so we should check that the catalog implements SupportsNamespace first.

Reply:
done.